app/src/main/java/org/schabi/newpipe/RouterActivity.java

```diff
@@ -154,6 +156,28 @@ protected void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         Icepick.restoreInstanceState(this, savedInstanceState);
 
```

Comment:
You should add some more comments to the code explaining what it is doing and maybe why it is done that way to make it easier for reviewers and other contributors. For example, it may be helpful to explain what additional information is added to the `searchIntent`. It may also be good to add some comments above the code to explain what feature it relates to.